### PR TITLE
[Testing] Add missing functional test covering matchExpressions of Presets

### PR DESCRIPTION
**Pull request by stu-gott**
```release-note
NONE
```

### CODE REVIEW

**Comment by stu-gott** [comment]
No changes. Just rebased.

**Comment by stu-gott** [comment]
ci test please

File: tests/vmipreset_test.go [hunk 1]
```diff
@@ -345,9 +363,8 @@ var _ = Describe("[rfe_id:609][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			err := virtClient.RestClient().Post().Resource("virtualmachineinstancepresets").Namespace(tests.NamespaceTestDefault).Body(overridePreset).Do().Error()
 			Expect(err).ToNot(HaveOccurred())
 
-			By("Waiting for preset to be created")
-			getPreset(virtClient, overridePrefix)
-			Expect(err).ToNot(HaveOccurred())
+			// Give virt-api's cache time to sync before proceeding
+			time.Sleep(3 * time.Second)
```
Comment (davidvossel) [inline tests/vmipreset_test.go]: hey, obviously based on all my previous feedback I'm in favor of the sleep here. but out of curiosity, when you removed the "getPreset", did the test occasionally fail before the sleep was added?

File: tests/vmipreset_test.go [hunk 2]
```diff
@@ -309,6 +324,9 @@ var _ = Describe("[rfe_id:609][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			getPreset(virtClient, memoryPrefix)
```
Comment (davidvossel) [inline tests/vmipreset_test.go]: can we get rid of the "getPreset()" function here now?

File: tests/vmipreset_test.go (hunk 1 above)
Comment (stu-gott) [reply tests/vmipreset_test.go]: yes *cough*

File: tests/vmipreset_test.go (hunk 2 above)
Comment (stu-gott) [reply tests/vmipreset_test.go]: fixed

File: tests/vmipreset_test.go (hunk 1 above)
Comment (stu-gott) [reply tests/vmipreset_test.go]: Overall I really dislike blind sleeps, but you make a good point. Doing it this way makes it glaringly obvious that there's a hacky workaround afoot.

**Comment by stu-gott** [comment]
/hold

We should fix https://github.com/kubevirt/kubevirt/pull/2299 and update this PR before merging.

**Pull request merged**